### PR TITLE
[Merged by Bors] - keep failed finalized chains to avoid retries

### PR DESCRIPTION
**Pull request by divagant-martian**
## Issue Addressed

In very rare occasions we've seen most if not all our peers in a chain with which we don't agree. Purging these peers can take a very long time: number of retries of the chain. Meanwhile sync is caught in a loop trying the chain again and again. This makes it so that we fast track purging peers via registering the failed chain to prevent retrying for some time (30 seconds). Longer times could be dangerous since a chain can fail if a batch fails to download for example. In this case, I think it's still acceptable to fast track purging peers since they are nor providing the required info anyway

### CODE REVIEW

**Review by AgeManning — APPROVED** [review]
Nice looks good to me

File: common/lru_cache/src/time.rs [hunk 1]
```diff
@@ -31,53 +31,45 @@ where
         }
     }
 
-    // Inserts new elements and removes any expired elements.
+    // Inserts a new key. It purges first expired elements to do so.
```
Comment (pawanjay176) [inline common/lru_cache/src/time.rs]:
```suggestion
    // Inserts a new key. It first purges expired elements to do so.
```

Comment (pawanjay176) [reply common/lru_cache/src/time.rs]: (might be misunderstanding the comment here)

**Review by pawanjay176 — APPROVED** [review]
LGTM as well. Just minor nits

**Pull request closed without merging**